tree/ntuple/inc/ROOT/RNTupleProcessorEntry.hxx

```diff
@@ -0,0 +1,266 @@
+   bool FieldIsValid(ROOT::RFieldToken token) const { return fFieldValidities[token.fIndex]; }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Check whether a field is valid for reading.
+   ///
+   /// \param[in] fieldName The name of the field in the entry.
+   bool FieldIsValid(std::string_view fieldName) const
+   {
+      auto token = FindToken(fieldName);
+      if (!token)
+         throw RException(R__FAIL("invalid field name: " + std::string(fieldName)));
+      return FieldIsValid(*token);
+   }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Check whether the entry currently has a certain field.
+   ///
+   /// \param[in] fieldName The name of the field to check.
+   bool HasField(const std::string &fieldName) const
+   {
+      return fFieldName2Token.find(fieldName) != fFieldName2Token.end();
+   }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Find the name of a field from its token
+   ///
+   /// \param[in] token The token representing the field in the entry.
+   ///
+   /// \warning This function has linear complexity, only use it for more helpful error messages!
+   const std::string &FindFieldName(ROOT::RFieldToken token) const
+   {
+      assert(token.fSchemaId == fProtoModel->fSchemaId && token.fIndex < fValues.size());
+      for (const auto &[fieldName, index] : fFieldName2Token) {
+         if (index == token.fIndex) {
+            return fieldName;
+         }
+      }
+      // Should never happen, but avoid compiler warning about "returning reference to local temporary object".
+      static const std::string empty = "";
+      return empty;
+   }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Get the token representing the provided field in the entry.
+   ///
+   /// \param[in] fieldName The name of the field in the entry.
+   std::optional<ROOT::RFieldToken> FindToken(std::string_view fieldName) const
+   {
+      auto it = fFieldName2Token.find(std::string(fieldName));
+      if (it == fFieldName2Token.end()) {
+         return std::nullopt;
+      }
+      return ROOT::RFieldToken(it->second, fProtoModel->GetSchemaId());
+   }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Add a new field to the entry or update the value pointer for an existing field.
+   ///
+   /// \param[in] fieldName The name of the field to add or update.
+   /// \param[in] valuePtr Pointer to the value for the field to add or update.
+   ///
+   /// \return `true` if the field was added successfully or if was already present and its value has been updated,
+   /// `false` otherwise.
+   ///
+   /// A field will only be updated if `valuePtr` is not a `nullptr`.
+   bool AddOrUpdateField(const std::string &fieldName, std::shared_ptr<void> valuePtr)
```

Comment:
```suggestion
   bool AddOrUpdateField(std::string_view fieldName, std::shared_ptr<void> valuePtr)
```
(unless there is a good reason to force a `std::string`; `FindToken` itself accepts a `std::string_view` and will convert it to `std::string` for searching in `fFieldName2Token`...)

Reply:
Hopefully that is not a performance critical path, otherwise it seems 'wasteful' to pass an `std::string` via `std::string_view` to turn around and make a duplicate of the same `std::string` (unless there is some transformation in between).